core/src/main/java/org/apache/iceberg/encryption/BaseEncryptedOutputFile.java

```diff
@@ -24,10 +24,19 @@ class BaseEncryptedOutputFile implements EncryptedOutputFile {
 
   private final OutputFile encryptingOutputFile;
   private final EncryptionKeyMetadata keyMetadata;
+  private final OutputFile rawOutputFile;
 
   BaseEncryptedOutputFile(OutputFile encryptingOutputFile, EncryptionKeyMetadata keyMetadata) {
+    this(encryptingOutputFile, keyMetadata, encryptingOutputFile);
+  }
+
+  BaseEncryptedOutputFile(
+      OutputFile encryptingOutputFile,
+      EncryptionKeyMetadata keyMetadata,
+      OutputFile rawOutputFile) {
```

Comment:
Two output files looks really suspicious to me. Maybe we would want to use a different implementation of the API to avoid that. Shouldn't this create the encrypting output file rather than having it passed in? Or is that an artifact from how the encryption manager works?

Reply:
I was considering this when looking at #6884. `EncryptedOutputFile` definitely needs to be able to return the underlying stream for Parquet encryption, but that doesn't mean that `BaseEncryptedInputFile` necessarily has to be used here.

Overall, I'm fine with this but it does look odd to have a wrapper that can supply either the raw output file or an encrypted one.

Reply:
I've reverted the `BaseEncryptedOutputFile` class. There is a new implementation of the updated `EncryptedOutputFile`, residing inside the encryption manager that provides the native Parquet encryption.